legal-api/src/legal_api/services/authz.py

```diff
@@ -136,6 +136,26 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
                     'business': [BusinessBlocker.DEFAULT]
                 }
             },
+            'amalgamation': {
+                'regular': {
+                    'legalTypes': ['BEN', 'BC', 'ULC', 'CC'],
+                    'blockerChecks': {
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                },
+                'vertical': {
+                    'legalTypes': ['BEN', 'BC', 'ULC', 'CC'],
+                    'blockerChecks': {
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                },
+                'horizontal': {
+                    'legalTypes': ['BEN', 'BC', 'ULC', 'CC'],
+                    'blockerChecks': {
+                        'business': [BusinessBlocker.DEFAULT]
+                    }
+                }
```

Comment:
I think this should be more similar to `incorporationApplication` and `registration` for amalgamations as we are essentially creating a new business.

Any validation that blocks the amalgamation for let's say good standing and other things will take place in the legal api validation for the amalgamation filing.  These are my thoughts about this for now anyways...

Reply:
Agreed, sort of... It's also possible to start an amalgamation from one of the amalgamating companies, ie:

![image](https://github.com/bcgov/lear/assets/10002889/036aa583-6a7b-4187-b6ba-d95fce51e525)

While it's true that this amalgamating company must be in good standing (unless you are staff) to use it in the amalgamation filing, I don't know if it's required to block this action if the company is NIGS. Ask Yui.

Reply:
yeah, that's one of the thoughts I had too...the starting from a specific business scenario.

It kind of sounded like you can still carry on with the amalgamation filing and the submission of the filing would fail validation.  

But yes, I'll check with Yui.

Reply:
Conversation with Yui in teams:

![image](https://github.com/bcgov/lear/assets/6685223/dc4ebb0e-9e40-4782-a457-cce656773e95)

Reply:
I think allowable actions is actually quite tricky.

The `amalgamations` entry here needs to be able to support the scenario of blocking an existing business from starting an amalgamation filing based off of a bunch of blocker conditions that Yui mentioned in teams and the also the creation of a new business via the amalgamation.  

When we were dealing with `registration` and `incorporationApplication` we could always just assume it's a new business.

As such, I think we need to add some new elements and refactor code to support these two things.  

I'm kind of thinking we implement a really minimal version of allowable actions that just solves the scenario where we start the amalgamation from the amalgamate button from the business dashboard for now.  

And we cut another ticket to that implements everything we need.  

If we don't do this, I think the ticket will blow up.  

What do you think @severinbeauvais ?

Reply:
Yes to "do what you can now and create another ticket for later".

If it helps, an amalgamation always creates a new business.

Reply:
I was doing - add the `businessExist: False`
If then, I will update it like that

Reply:
yeah, keep it like what you had before with no businessExists.  

Just update what you had to include & exclude what I mentioned above.

We will deal with the other stuff like businessExists in #18784